### PR TITLE
Prevent crash with empty database

### LINKED ISSUE #1143: Crash when creating a new database
When creating a new database with develop HEAD, KeePassXC always crashes.

Bisected to:
```
eb6582266533a979a753327b7f222867b7d146c3 is the first bad commit
commit eb6582266533a979a753327b7f222867b7d146c3
Author: thez3ro <io@thezero.org>
Date:   Wed Sep 27 19:18:13 2017 +0200

    layout update

```

Ping @TheZ3ro

## Steps to Reproduce (for bugs)

1. Start KeePassXC
2. Create new database
3. Will crash immediately

## Debug Info

KeePassXC - Version 2.2.2
Revision: eb65822

Libraries:
- Qt 5.7.1
- libgcrypt 1.7.8

Operating system: Fedora 26 (Twenty Six)
CPU architecture: x86_64
Kernel: linux 4.13.5-200.fc26.x86_64

Enabled extensions: None

### PR DESCRIPTION
**Pull request by TheZ3ro**
Fix #1143 
Regression in commit eb6582266533a979a753327b7f222867b7d146c3

## Description

Hide the DetailsView if there are no entry instead of crashing

## How has this been tested?

Manually

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by TheZ3ro** [comment]
@frostasm can you test this?

**Comment by frostasm** [comment]
yes

File: src/gui/DetailsWidget.cpp [hunk 1]
```diff
@@ -309,9 +309,17 @@ void DetailsWidget::setDatabaseMode(DatabaseWidget::Mode mode)
...
-            getSelectedGroup(m_currentGroup);
+            if (m_currentGroup) {
+                getSelectedGroup(m_currentGroup);
+            } else {
+                hideDetails();
+            }
         } else {
-            getSelectedEntry(m_currentEntry);
+            if (m_currentEntry) {
+                getSelectedEntry(m_currentEntry);
```
Comment (louib) [inline src/gui/DetailsWidget.cpp]: @TheZ3ro shouldn't this logic go in `getSelectedEntry` (and `getSelectedGroup`) instead?

Comment (TheZ3ro) [reply src/gui/DetailsWidget.cpp]: Mmmmh yeah, why not

Comment (frostasm) [reply src/gui/DetailsWidget.cpp]: @TheZ3ro why functions start with "get*", why not with "set*" or "update*"?

**Comment by TheZ3ro** [comment]
@louib fixed

**Comment by frostasm** [comment]
@TheZ3ro maybe these changes are also needed in the 2.2.3 branch?

**Comment by TheZ3ro** [comment]
@frostasm the regression commit comes from #879 that has been merged in develop branch for future 2.3.0 release, 2.2.* releases are fine and not affected ;)

**Comment by frostasm** [comment]
@TheZ3ro thanks. My mistake, I was inattentive ;)

**Pull request merged**